polaris-react/src/components/Popover/components/PopoverOverlay/tests/PopoverOverlay.test.tsx

```diff
@@ -277,18 +273,116 @@ describe('<PopoverOverlay />', () => {
       </PopoverOverlay>,
     );
 
-    const target = popoverOverlay.find(TextField)!.find('svg')!.domNode;
-    // eslint-disable-next-line import/no-deprecated
-    const clickEventListener = popoverOverlay.find(EventListener, {
-      event: 'click',
-    })!;
+    const target = popoverOverlay.find(TextField)!.find('svg')!.domNode!;
+    const clickEvent = new MouseEvent('click', {bubbles: true});
+
+    target.dispatchEvent(clickEvent);
 
-    clickEventListener!.trigger('handler', {
-      target,
-    });
     expect(spy).not.toHaveBeenCalled();
   });
 
+  it('does not call the onClose callback when another overlay is clicked if preventCloseOnChildOverlayClick is true', () => {
+    const spy = jest.fn();
+
+    const popoverOverlayAndPopover = mountWithApp(
+      <div>
+        <PopoverOverlay
+          active
+          id="PopoverOverlay-1"
+          activator={activator}
+          onClose={spy}
+          preventCloseOnChildOverlayClick
+        >
+          <button id="popover 1 button" />
+        </PopoverOverlay>
+        <Popover
+          active
+          fullWidth
+          activator={<div>Activator</div>}
+          onClose={jest.fn()}
+        >
+          <button id="popover 2 button" />
+        </Popover>
+      </div>,
+    );
+
+    const popover2Button = popoverOverlayAndPopover.find('button', {
+      id: 'popover 2 button',
+    })!.domNode!;
+    const clickEvent = new MouseEvent('click', {bubbles: true});
+
+    popover2Button.dispatchEvent(clickEvent);
+
+    expect(spy).not.toHaveBeenCalled();
+  });
+
+  describe('deleting descendant elements', () => {
+    const DeleteButton = () => {
+      const [show, setShow] = React.useState(true);
+
+      return (
+        <>{show && <button onClick={() => setShow(!show)}>Delete</button>}</>
+      );
```

Comment:
```suggestion
      return show && <button onClick={() => setShow(!show)}>Delete</button>;
```

Reply:
React and the linter won't let me do this. React requires enclosing it in fragment tags, and if I try to one-line that the linter changes it to what I have.

Reply:
That's because React is trying to render `false` below, so you'll need a ternary:

```suggestion
      return show ? (
        <button onClick={() => setShow(!show)}>Delete</button>
      ) : null;
```